presto-spi/src/main/java/com/facebook/presto/spi/block/RunLengthEncodedBlock.java

```diff
@@ -309,55 +309,55 @@ private void checkReadablePosition(int position)
     @Override
     public byte getByteUnchecked(int internalPosition)
     {
-        return value.getByteUnchecked(0);
+        return value.getByteUnchecked(internalPosition);
```

Comment:
This should be squashed into commit "Add UncheckedBlock" right? Also I don't quite understand this change -- it should be `value.getByteUnchecked(0)` right? Otherwise you are "penetrating" the position into the underlying value block, which doesn't sounds the right behavior for `getXUnchecked`